src/common/types/generic.ts

```
@@ -55,3 +55,31 @@ export const ArrayFromString = z.preprocess(
+export type MaxLengthString<N extends number> = string & { readonly [__maxLength]: N };
+
+type ValidateMaxLength<S extends string, N extends number> =
+  string extends S
+  ? never
+  : CheckLength<S, N> extends true
+  ? S
+  : never;
+
+export const maxLength = <const S extends string, N extends number>(
+  s: ValidateMaxLength<S, N>,
+  _n: N
```

ESLint

[error] Insert `,`

(prettier/prettier)